### ISSUE TITLE
Support: Invalid JSON input through the Populate from... Freetext Import

### ISSUE THREAD

**Issue by caiquebarqueta** [MISP/MISP#9991]
### Support Questions

I'm having trouble installing MISP with Populate From... with a list of hashes and free IPs to import into the event.

I'm getting the following error:

```
2024-10-16 19:48:31 Error: [HttpException] Invalid JSON input. Make sure that the JSON input is a correctly formatted JSON string. This request has been blocked to avoid an unfiltered request.
Request URL: /events/saveFreeText/4
Stack Trace:
#0 /var/www/MISP/app/Controller/EventsController.php(4226): AppController->_jsonDecode()
#1 [internal function]: EventsController->saveFreeText()
#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(500): ReflectionMethod->invokeArgs()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction()
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#5 /var/www/MISP/app/webroot/index.php(99): Dispatcher->dispatch()
#6 {main}
```

### MISP version

2.5

### Operating System

Ubuntu

### Operating System version

24.04

### PHP version

8.3

### Browser

Firefox, Chrome, any. 

### Relevant log output

```shell
2024-10-16 19:48:31 Error: [HttpException] Invalid JSON input. Make sure that the JSON input is a correctly formatted JSON string. This request has bee
Request URL: /events/saveFreeText/4
Stack Trace:
#0 /var/www/MISP/app/Controller/EventsController.php(4226): AppController->_jsonDecode()
#1 [internal function]: EventsController->saveFreeText()
#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(500): ReflectionMethod->invokeArgs()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction()
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#5 /var/www/MISP/app/webroot/index.php(99): Dispatcher->dispatch()
#6 {main}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by iglocska** [comment]
Hmm, I can't seem to reproduce it, but I remember there being a bug with
the freetext import on 2.5. Could you git pull just in case to update to
the latest commit? We'll be doing a 2.5.1 release today to have the fixes
in a tagged version if that doesn't work for you.

**Comment by caiquebarqueta** [comment]
I did the update, but it still persists.

I'm also trying to check what it could be.


```
2024-10-18 13:09:10 Warning: Warning (2): Undefined array key "MispAttribute" in [/var/www/MISP/app/Controller/EventsController.php, line 4224]
Trace:
ErrorHandler::handleError() - APP/Lib/cakephp/lib/Cake/Error/ErrorHandler.php, line 232
EventsController::saveFreeText() - APP/Controller/EventsController.php, line 4224
ReflectionMethod::invokeArgs() - [internal], line ??
Controller::invokeAction() - APP/Lib/cakephp/lib/Cake/Controller/Controller.php, line 500
Dispatcher::_invoke() - APP/Lib/cakephp/lib/Cake/Routing/Dispatcher.php, line 193
Dispatcher::dispatch() - APP/Lib/cakephp/lib/Cake/Routing/Dispatcher.php, line 167
[main] - APP/webroot/index.php, line 107

2024-10-18 13:09:10 Warning: Warning (2): Trying to access array offset on null in [/var/www/MISP/app/Controller/EventsController.php, line 4226]
Trace:
ErrorHandler::handleError() - APP/Lib/cakephp/lib/Cake/Error/ErrorHandler.php, line 232
EventsController::saveFreeText() - APP/Controller/EventsController.php, line 4226
ReflectionMethod::invokeArgs() - [internal], line ??
Controller::invokeAction() - APP/Lib/cakephp/lib/Cake/Controller/Controller.php, line 500
Dispatcher::_invoke() - APP/Lib/cakephp/lib/Cake/Routing/Dispatcher.php, line 193
Dispatcher::dispatch() - APP/Lib/cakephp/lib/Cake/Routing/Dispatcher.php, line 167
[main] - APP/webroot/index.php, line 107

2024-10-18 13:09:10 Error: [HttpException] Invalid JSON input. Make sure that the JSON input is a correctly formatted JSON string. This request has been blocked to avoid an unfiltered request.
Request URL: /events/saveFreeText/4
Stack Trace:
#0 /var/www/MISP/app/Controller/EventsController.php(4226): AppController->_jsonDecode()
#1 [internal function]: EventsController->saveFreeText()
#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(500): ReflectionMethod->invokeArgs()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction()
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#5 /var/www/MISP/app/webroot/index.php(107): Dispatcher->dispatch()
#6 {main}
2024-10-18 13:09:13 Error: Blackhole exception when accessing /events/saveFreeText/4 (isRest: 0, action: saveFreeText, unlockedActions: ["viewEventAttributes"]): CSRF token mismatch
2024-10-18 13:09:13 Error: [BadRequestException] The request has been black-holed
Request URL: /events/saveFreeText/4
Stack Trace:
#0 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(831): AppController->blackhole()
#1 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(351): SecurityComponent->_callback()
#2 /var/www/MISP/app/Controller/Component/BetterSecurityComponent.php(22): SecurityComponent->blackHole()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(255): BetterSecurityComponent->blackHole()
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Utility/ObjectCollection.php(129): SecurityComponent->startup()
#5 /var/www/MISP/app/Lib/Tools/BetterCakeEventManager.php(23): ObjectCollection->trigger()
#6 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(688): BetterCakeEventManager->dispatch()
#7 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(189): Controller->startupProcess()
#8 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#9 /var/www/MISP/app/webroot/index.php(107): Dispatcher->dispatch()
#10 {main}
```

![image](https://github.com/user-attachments/assets/ab5b4540-5e54-4987-b5db-0ae10f75ee9e)

**Comment by samitainio** [comment]
@iglocska I am also seeing these errors in the `/var/www/MISP/app/tmp/logs/error.log` file. I attempted to use the Freetext Import and once you see the Freetext Import Results page and click the Submit attributes button these errors appeared to the logs.

MISP version:  v2.5.1 (5c4e205f2a224227123833ccbd46d040a05cd66c)
PHP version: 8.3.6

```
2024-10-20 08:51:53 Error: Blackhole exception when accessing /events/saveFreeText/2479 (isRest: 0, action: saveFreeText, unlockedActions: ["viewEventAttributes"]): CSRF token mismatch
2024-10-20 08:51:53 Error: [BadRequestException] The request has been black-holed
Request URL: /events/saveFreeText/2479
Stack Trace:
#0 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(831): AppController->blackhole()
#1 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(351): SecurityComponent->_callback()
#2 /var/www/MISP/app/Controller/Component/BetterSecurityComponent.php(22): SecurityComponent->blackHole()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(255): BetterSecurityComponent->blackHole()
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Utility/ObjectCollection.php(129): SecurityComponent->startup()
#5 /var/www/MISP/app/Lib/Tools/BetterCakeEventManager.php(23): ObjectCollection->trigger()
#6 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(688): BetterCakeEventManager->dispatch()
#7 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(189): Controller->startupProcess()
#8 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#9 /var/www/MISP/app/webroot/index.php(107): Dispatcher->dispatch()
#10 {main}

2024-10-20 08:59:59 Error: Blackhole exception when accessing /events/saveFreeText/2479 (isRest: 0, action: saveFreeText, unlockedActions: ["viewEventAttributes"]): CSRF token mismatch
2024-10-20 08:59:59 Error: [BadRequestException] The request has been black-holed
Request URL: /events/saveFreeText/2479
Stack Trace:
#0 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(831): AppController->blackhole()
#1 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(351): SecurityComponent->_callback()
#2 /var/www/MISP/app/Controller/Component/BetterSecurityComponent.php(22): SecurityComponent->blackHole()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(255): BetterSecurityComponent->blackHole()
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Utility/ObjectCollection.php(129): SecurityComponent->startup()
#5 /var/www/MISP/app/Lib/Tools/BetterCakeEventManager.php(23): ObjectCollection->trigger()
#6 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(688): BetterCakeEventManager->dispatch()
#7 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(189): Controller->startupProcess()
#8 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#9 /var/www/MISP/app/webroot/index.php(107): Dispatcher->dispatch()
#10 {main}

2024-10-20 09:00:11 Error: Blackhole exception when accessing /events/freeTextImport/2479 (isRest: 0, action: freeTextImport, unlockedActions: ["viewEventAttributes"]): CSRF token mismatch
2024-10-20 09:00:11 Error: [BadRequestException] The request has been black-holed
Request URL: /events/freeTextImport/2479?_=1729414205728
Stack Trace:
#0 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(831): AppController->blackhole()
#1 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(351): SecurityComponent->_callback()
#2 /var/www/MISP/app/Controller/Component/BetterSecurityComponent.php(22): SecurityComponent->blackHole()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(255): BetterSecurityComponent->blackHole()
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Utility/ObjectCollection.php(129): SecurityComponent->startup()
#5 /var/www/MISP/app/Lib/Tools/BetterCakeEventManager.php(23): ObjectCollection->trigger()
#6 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(688): BetterCakeEventManager->dispatch()
#7 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(189): Controller->startupProcess()
#8 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#9 /var/www/MISP/app/webroot/index.php(107): Dispatcher->dispatch()
#10 {main}

2024-10-20 09:00:40 Warning: Warning (2): Undefined array key "MispAttribute" in [/var/www/MISP/app/Controller/EventsController.php, line 4224]
Trace:
ErrorHandler::handleError() - APP/Lib/cakephp/lib/Cake/Error/ErrorHandler.php, line 232
EventsController::saveFreeText() - APP/Controller/EventsController.php, line 4224
ReflectionMethod::invokeArgs() - [internal], line ??
Controller::invokeAction() - APP/Lib/cakephp/lib/Cake/Controller/Controller.php, line 500
Dispatcher::_invoke() - APP/Lib/cakephp/lib/Cake/Routing/Dispatcher.php, line 193
Dispatcher::dispatch() - APP/Lib/cakephp/lib/Cake/Routing/Dispatcher.php, line 167
[main] - APP/webroot/index.php, line 107

2024-10-20 09:00:40 Warning: Warning (2): Trying to access array offset on null in [/var/www/MISP/app/Controller/EventsController.php, line 4226]
Trace:
ErrorHandler::handleError() - APP/Lib/cakephp/lib/Cake/Error/ErrorHandler.php, line 232
EventsController::saveFreeText() - APP/Controller/EventsController.php, line 4226
ReflectionMethod::invokeArgs() - [internal], line ??
Controller::invokeAction() - APP/Lib/cakephp/lib/Cake/Controller/Controller.php, line 500
Dispatcher::_invoke() - APP/Lib/cakephp/lib/Cake/Routing/Dispatcher.php, line 193
Dispatcher::dispatch() - APP/Lib/cakephp/lib/Cake/Routing/Dispatcher.php, line 167
[main] - APP/webroot/index.php, line 107

2024-10-20 09:00:40 Error: [HttpException] Invalid JSON input. Make sure that the JSON input is a correctly formatted JSON string. This request has been blocked to avoid an unfiltered request.
Request URL: /events/saveFreeText/2479
Stack Trace:
#0 /var/www/MISP/app/Controller/EventsController.php(4226): AppController->_jsonDecode()
#1 [internal function]: EventsController->saveFreeText()
#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(500): ReflectionMethod->invokeArgs()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction()
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#5 /var/www/MISP/app/webroot/index.php(107): Dispatcher->dispatch()
#6 {main}
```

**Comment by SAIKATDASGITHUB** [comment]
Hi,

Anybody found a solution for this?

@iglocska I am also getting the same error - Cross site request forgery protection of MISP

This error I am getting while adding a new organization and also while adding a new sync server.

Version - 2.5.1
PHP - 8.3

Error Log:

2024-10-23 10:26:52 Error: Blackhole exception when accessing /admin/organisations/add (isRest: 0, action: admin_add, unlockedActions: []): The request has >
2024-10-23 10:26:52 Error: [BadRequestException] The request has been black-holed
Request URL: /admin/organisations/add

Stack Trace:
#0 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(831): AppController->blackhole()
#1 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(351): SecurityComponent->_callback()
#2 /var/www/MISP/app/Controller/Component/BetterSecurityComponent.php(22): SecurityComponent->blackHole()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(255): BetterSecurityComponent->blackHole()
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Utility/ObjectCollection.php(129): SecurityComponent->startup()
#5 /var/www/MISP/app/Lib/Tools/BetterCakeEventManager.php(23): ObjectCollection->trigger()
#6 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(683): BetterCakeEventManager->dispatch()
#7 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(189): Controller->startupProcess()
#8 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#9 /var/www/MISP/app/webroot/index.php(107): Dispatcher->dispatch()
#10 {main}

**Comment by m4d-sc13nt1st** [comment]
Also getting this error when adding a list of IP addresses via the Freetext Import module:

MISP version: v2.5.1
PHP version: 8.3.6

/var/www/MISP/app/tmp/logs:

```
2024-10-24 13:19:58 Error: Blackhole exception when accessing /events/saveFreeText/66286 (isRest: 0, action: saveFreeText, unlockedActions: ["viewEventAttributes"]): CSRF token mismatch
2024-10-24 13:19:58 Error: [BadRequestException] The request has been black-holed
Request URL: /events/saveFreeText/66286
Stack Trace:
#0 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(831): AppController->blackhole()
#1 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(351): SecurityComponent->_callback()
#2 /var/www/MISP/app/Controller/Component/BetterSecurityComponent.php(22): SecurityComponent->blackHole()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(255): BetterSecurityComponent->blackHole()
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Utility/ObjectCollection.php(129): SecurityComponent->startup()
#5 /var/www/MISP/app/Lib/Tools/BetterCakeEventManager.php(23): ObjectCollection->trigger()
#6 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(688): BetterCakeEventManager->dispatch()
#7 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(189): Controller->startupProcess()
#8 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#9 /var/www/MISP/app/webroot/index.php(107): Dispatcher->dispatch()
#10 {main}
```

After temporarily setting Security.disable_form_security to true, the error changes:

```
2024-10-24 13:22:01 Error: [HttpException] Invalid JSON input. Make sure that the JSON input is a correctly formatted JSON string. This request has been blocked to avoid an unfiltered request.
Request URL: /events/saveFreeText/66286
Stack Trace:
#0 /var/www/MISP/app/Controller/EventsController.php(4224): AppController->_jsonDecode()
#1 [internal function]: EventsController->saveFreeText()
#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(500): ReflectionMethod->invokeArgs()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction()
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#5 /var/www/MISP/app/webroot/index.php(107): Dispatcher->dispatch()
#6 {main}
```

**Update**: This only happens when using Firefox. Import works fine in Chrome.

**Comment by caiquebarqueta** [comment]
I intend to explore it further this week to identify more details as to why the problem exists.

**Comment by samitainio** [comment]
@caiquebarqueta Atleast for me this issue has been fixed in the bb0909a37a7bd727d35f7a60ee9a6ff10bc99852 commit.

**Comment by caiquebarqueta** [comment]
I really didn't pay attention to that part of the commit. Thanks for your support. :)
